protostar/cli/network_command_mixin.py

```diff
@@ -0,0 +1,80 @@
+from logging import Logger
+from typing import Any
+
+from starknet_py.net.gateway_client import GatewayClient
+
+from protostar.cli import Command
+from protostar.protostar_exception import ProtostarException
+from protostar.starknet_gateway import NetworkConfig
+from protostar.starknet_gateway.network_config import is_legacy_network_name, NETWORKS
+
+GATEWAY_URL_ARG_NAME = "gateway-url"
+NETWORK_ARG_NAME = "network"
+CHAIN_ID_ARG_NAME = "chain-id"
+
+
+class NetworkCommandMixin:
+    @property
+    def network_arguments(self):
+        return [
+            Command.Argument(
+                name=GATEWAY_URL_ARG_NAME,
+                description="The URL of a StarkNet gateway. It is required unless `--network` is provided.",
+                type="str",
+            ),
+            Command.Argument(
+                name=CHAIN_ID_ARG_NAME,
+                description="The chain id. It is required unless `--network` is provided.",
+                type="int",
+            ),
+            Command.Argument(
+                name=NETWORK_ARG_NAME,
+                short_name="n",
+                description=(
+                    "\n".join(
+                        [
+                            "The name of the StarkNet network.",
+                            "It is required unless `--gateway-url` is provided.",
+                            "",
+                            "Supported StarkNet networks:",
+                        ]
+                        + [f"- `{n}`" for n in NETWORKS]
+                    )
+                ),
+                type="str",
+            ),
+        ]
+
+    @staticmethod
+    def validate_network_command_args(args, logger: Logger):
+        if args.network is None and args.gateway_url is None:
+            raise ProtostarException(
+                f"Argument `{GATEWAY_URL_ARG_NAME}` or `{NETWORK_ARG_NAME}` is required"
+            )
+
+        # TODO(arcticae): Remove in the future version, with the legacy names formats support
+        if args.network and is_legacy_network_name(args.network):
+            logger.warning(
+                f"{args.network} is a legacy network name parameter and it won't be supported in future versions"
+            )
+
+        if args.gateway_url and not args.chain_id:
+            raise ProtostarException(
+                f"Argument `{CHAIN_ID_ARG_NAME}` is required when `{GATEWAY_URL_ARG_NAME}` is provided"
+            )
+
+    @staticmethod
+    def get_network_config(args: Any, logger: Logger) -> NetworkConfig:
+        NetworkCommandMixin.validate_network_command_args(args, logger)
+        return NetworkConfig.build(
+            args.gateway_url, args.network, chain_id=args.chain_id
+        )
+
+    @staticmethod
+    def get_gateway_client(args: Any, logger: Logger) -> GatewayClient:
```

Comment:
logger was supposed to be made static from what i've heard? So that's not going to be an issue anymore soon.

Reply:
What solution do you propose here to make this logic modular? This solution quite nicely decouples this logic from commands (only command needs to know about it's components, not the other way around)

Reply:
`log_color_provider` can be used as a global variable.

Logger however is going to be abstracted and adapted to Protostar needs (to avoid interchangeably using logger and `print` function).

`getLogger()` would make sense if it would be used for internal needs, not as a way of presenting results to the user.

Reply:
> What solution do you propose here to make this logic modular? This solution quite nicely decouples this logic from commands (only command needs to know about it's components, not the other way around)

Composition

Reply:
done